frontend/packages/pipelines-plugin/src/components/shared/workspaces/WorkspaceResourceLinkList.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { ResourceLink } from '@console/internal/components/utils';
+import { PersistentVolumeClaimModel, ConfigMapModel, SecretModel } from '@console/internal/models';
+import {
+  PipelineRunWorkspace,
+  VolumeTypeConfigMaps,
+  VolumeTypePVC,
+  VolumeTypeSecret,
+} from '../../../types';
+import VolumeClaimTemplatesLink from './VolumeClaimTemplateLink';
+
+export interface WorkspaceResourceLinkListProps {
+  workspaces: PipelineRunWorkspace[];
+  namespace: string;
+  ownerResourceName: string;
+  ownerResourceKind?: string;
+}
+
+const WorkspaceResourceLinkList: React.FC<WorkspaceResourceLinkListProps> = ({
+  workspaces,
+  namespace,
+  ownerResourceName,
+  ownerResourceKind,
+}) => {
+  const { t } = useTranslation();
+
+  if (!workspaces || workspaces.length === 0) return null;
+
+  return (
+    <dl>
+      <dt>{t('pipelines-plugin~Workspaces')}</dt>
+      <dd>
+        {workspaces.map((workspace) => {
+          if (workspace.persistentVolumeClaim) {
+            const persistentVolumeClaim = workspace.persistentVolumeClaim as VolumeTypePVC;
+            const displayName = `${persistentVolumeClaim.claimName} (${workspace.name})`;
+            return (
+              <ResourceLink
+                key={workspace.name}
+                name={persistentVolumeClaim.claimName}
+                namespace={namespace}
+                kind={PersistentVolumeClaimModel.kind}
+                displayName={displayName}
+              />
+            );
+          }
+          if (workspace.configMap) {
+            const configMap = workspace.configMap as VolumeTypeConfigMaps;
+            const displayName = `${configMap.name} (${workspace.name})`;
+            return (
+              <ResourceLink
+                key={workspace.name}
+                name={configMap.name}
+                namespace={namespace}
+                kind={ConfigMapModel.kind}
+                displayName={displayName}
+              />
+            );
+          }
+          if (workspace.secret) {
+            const secret = workspace.secret as VolumeTypeSecret;
+            const displayName = `${secret.secretName} (${workspace.name})`;
+            return (
+              <ResourceLink
+                key={workspace.name}
+                name={secret.secretName}
+                kind={SecretModel.kind}
+                namespace={namespace}
+                displayName={displayName}
+              />
+            );
+          }
+          return <div key={workspace.name}>{workspace.name}</div>;
+        })}
+        <VolumeClaimTemplatesLink
+          namespace={namespace}
+          ownerResourceName={ownerResourceName}
+          ownerResourceKind={ownerResourceKind}
+        />
```

Comment:
@abhinandan13jan Please test before you request a review... You are double rendering...

![image](https://user-images.githubusercontent.com/8126518/109858553-f7555480-7c29-11eb-9c83-7d2b229f3b26.png)

Or if you prefer, with my two `volumeClaimTemplate` example from before:
![image](https://user-images.githubusercontent.com/8126518/109858861-59ae5500-7c2a-11eb-896b-b884a03a4cc7.png)

Reply:
My PLR YAML, completely missed it. I've updated and tested against these now. Thanks a ton for the check

![Screenshot from 2021-03-04 02-48-29](https://user-images.githubusercontent.com/24852534/109876502-e4796900-7c97-11eb-893d-94264c08a569.png)
![Screenshot from 2021-03-04 02-49-04](https://user-images.githubusercontent.com/24852534/109876511-e6432c80-7c97-11eb-8007-e01e10b5f317.png)